crates/constraint-vm/src/access.rs

```diff
@@ -192,23 +192,35 @@ pub(crate) fn decision_var_len(solution: SolutionAccess, stack: &mut Stack) -> O
     })
 }
 
-/// `Access::MutKeysLen` implementation.
-pub(crate) fn mut_keys_len(solution: SolutionAccess, stack: &mut Stack) -> OpResult<()> {
-    stack.push(
-        solution
-            .mutable_keys
-            .len()
-            .try_into()
-            .map_err(|_| AccessError::SolutionDataOutOfBounds)?,
+/// `Access::MutKeys` implementation.
+pub(crate) fn push_mut_keys(solution: SolutionAccess, stack: &mut Stack) -> OpResult<()> {
+    let length = solution
+        .mutable_keys
+        .iter()
+        .map(|i| i.len())
+        .sum::<usize>()
+        .checked_add(solution.mutable_keys.len());
+    let total_length = length.and_then(|i| Word::try_from(i).ok()).ok_or(
+        AccessError::StateMutationsLengthTooLarge(solution.mutable_keys.len()),
     )?;
-    Ok(())
-}
-
-pub(crate) fn mut_keys_contains(solution: SolutionAccess, stack: &mut Stack) -> OpResult<()> {
-    let found = stack.pop_len_words::<_, bool, crate::error::OpError>(|words| {
-        Ok(solution.mutable_keys.contains(words))
-    })?;
-    stack.push(Word::from(found))?;
+    let iter = solution
+        .mutable_keys
+        .iter()
+        .filter(|key| {
+            // This should never happen because keys have
+            // a capped size but we don't want to return
+            // an error because it would require collecting
+            // into a vec. But we also don't want to panic.
+            key.len() <= (Word::MAX as usize)
```

Comment:
Hmm I'm not a fan of putting an assert here because it will panic the vm if ever triggered. The problem is this can't occur if the vm is used correctly (the validation checks are run on the input before calling this function but it could in theory happen if someone just uses the vm directly. It's still very unlikely as it would be insane to have `i64::MAX` sized keys.
Maybe the correct thing to do is to return a result but if we are going to push things onto the stack individually I might expose `reserve` so we can reserve enough capacity so there's no allocations when looping over push.